lib/question.js

```diff
@@ -349,10 +349,19 @@ module.exports = {
             },
             (callback) => { // if we have a workspace, get the files
                 if (workspace_id == null) return callback(null);
-                util.callbackify(workspace.getGradedFiles)(workspace_id, (err, result) => {
+                sqldb.queryZeroOrOneRow(sql.select_workspace_graded_files, {workspace_id}, (err, result) => {
                     if (ERR(err, callback)) return;
-                    zipPath = result;
-                    callback(null);
+                    let workspace_graded_files = [];
+                    if (result.rowCount > 0) {
+                        workspace_graded_files = result.rows[0].workspace_graded_files;
+                    }
```

Comment:
I think it's safer to do the following, so that we will get `[]` even if we have a row but `gradedFiles` is `NULL`.

```suggestion
                    const workspace_graded_files = result.rows[0].workspace_graded_files || [];
```